### ISSUE TITLE
Refactor filter API

### ISSUE THREAD

**Issue by whitepaperclip** [smpallen99/ex_admin#327]
As discussed in PR #312:

`filter [posts: [label: "Posts", order_by: [asc: :name]]]`

**Comment by whitepaperclip** [comment]
@smpallen99 possible to assign this to me?

**Comment by smpallen99** [comment]
@whitepaperclip Just had request  for a feature in the filters page. Should see an issue coming in soon. Someone wants to override association filter for large data so they can search on binary foreign_key instead of using select control. 

I'm thinking that we add an option to allow text/integer search on key instead of select. 

Let me know if you want to tackle this one too.

**Comment by whitepaperclip** [comment]
@smpallen99 Yep, that should go into the refactor as well. Feel free to assign.

**Comment by gregnar** [comment]
@smpallen99 @whitepaperclip I just submitted the issue in question, #328 

Thanks!

**Comment by whitepaperclip** [comment]
@gregnar Just submitted the refactored API. Should fix this.

**Comment by smpallen99** [comment]
will look at it when I get back from vacation in a couple days
